### PR TITLE
fix(Core/Spells): Implemented SPELL_ATTR3_NO_PROC_EQUIP_REQUIREMENT.

### LINKED ISSUE #6559: [warrior] Deep Wounds talent does not proc off Ranged Attack crits
#### WHAT CLIENT DO YOU PLAY ON?

- [ ] enGB
- [x] enUS
- [ ] other (specify)

##### FACTION

- [x] Alliance
- [x] Horde

##### CONTENT PHASE:

- [x] Generic
- [ ] 1-19
- [ ] 20-29
- [ ] 30-39
- [ ] 40-49
- [ ] 50-59

Original report: https://github.com/chromiecraft/chromiecraft/issues/969

##### CURRENT BEHAVIOUR:

Getting a critical with a Ranged attack does not currently proc the warrior talent called deep wounds
https://wowgaming.altervista.org/aowow/?spell=12867

Video demo: https://youtu.be/6eomj8vePZA

##### EXPECTED BLIZZLIKE BEHAVIOUR:

''Your critical strikes cause the opponent to bleed, dealing 48% of your melee weapon's average damage over 6 sec.''
Every single critical, regardless of its source, weapon swing, thunderclap, shield slam, heroic thrown or a **ranged attack** should proc the deep wounds talent on your target.
Your main hand should be used to calculate how much damage it should do.

##### SOURCE: 
https://www.mmo-champion.com/threads/623902-Question-Deep-Wounds
Screenshot_24

##### STEPS TO REPRODUCE THE PROBLEM:

1. get a warrior in his mid 20s
2. learn deep wounds talent, get your critical chance really high by equiping some OP wotlk quest gear with no gear req
3. use shoot skill against a dummy
4. it will crit, but no deep wounds will be applied to your target.

##### EXTRA NOTES:

https://wowwiki-archive.fandom.com/wiki/Deep_Wounds no changes made saying it should proc off melee abilities only, there was even a fix saying it should proc off heroic thrown.

##### AC HASH/COMMIT:

https://github.com/chromiecraft/azerothcore-wotlk/commit/34da0cda5195dede48cb0406b23248330d6249cb

##### OPERATING SYSTEM:

Ubuntu 20.04

##### MODULES:

- [mod-ah-bot](https://github.com/azerothcore/mod-ah-bot)
- [mod-cfbg](https://github.com/azerothcore/mod-cfbg)
- [mod-desertion-warnings](https://github.com/azerothcore/mod-desertion-warnings)
- [mod-duel-reset](https://github.com/azerothcore/mod-duel-reset)
- [mod-eluna-lua-engine](https://github.com/azerothcore/mod-eluna-lua-engine)
- [mod-ip-tracker](https://github.com/azerothcore/mod-ip-tracker),
- [mod-low-level-arena](https://github.com/azerothcore/mod-low-level-arena)
- [mod-multi-client-check](https://github.com/azerothcore/mod-multi-client-check)
- [mod-pvp-titles](https://github.com/azerothcore/mod-pvp-titles)
- [mod-queue-list-cache](https://github.com/azerothcore/mod-queue-list-cache)
- [mod-server-auto-shutdown](https://github.com/azerothcore/mod-server-auto-shutdown)
- [lua-CarbonCopy](https://github.com/55Honey/Acore_CarbonCopy)
- [lua-LevelUpReward](https://github.com/55Honey/Acore_LevelUpReward)
- [lua-send-and-bind](https://github.com/55Honey/Acore_SendAndBind)
- [lua-Zonecheck](https://github.com/55Honey/acore_Zonecheck)

##### OTHER CUSTOMIZATIONS:

None.

##### SERVER:

ChromieCraft

Confirmed and reproduced on a local AC server (AC HASH/COMMIT: 2a2e54d) :

WoWScrnShot_062421_205538

<bountysource-plugin>

---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/99451419-warrior-deep-wounds-talent-does-not-proc-off-ranged-attack-crits?utm_campaign=plugin&utm_content=tracker%2F40032087&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F40032087&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by UltraNix**
Fixed #6559.

## Changes Proposed:
-  Implemented SPELL_ATTR3_NO_PROC_EQUIP_REQUIREMENT

## Issues Addressed:

- Closes #6559
- Closes https://github.com/chromiecraft/chromiecraft/issues/969

## Tests Performed:

- Tested ingame.

## How to Test the Changes:

1. get a warrior in his mid 20s
2. learn deep wounds talent, get your critical chance really high by equiping some OP wotlk quest gear with no gear req
3. use shoot skill against a dummy

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Review by Kitzunu — APPROVED** [review]
nice

**Comment by pklloveyou** [comment]
Shooting does not trigger this spell, any other spell can trigger it

**Comment by UltraNix** [comment]
@pklloveyou Should be okay now.

**Comment by pklloveyou** [comment]
OK, I'll try

**Comment by pklloveyou** [comment]
Tested

File: src/server/game/Entities/Unit/Unit.cpp [hunk 1]
```diff
@@ -16673,8 +16673,10 @@ bool Unit::IsTriggeredAtSpellProcEvent(Unit* victim, Aura* aura, SpellInfo const
 
     // Xinef: additional check for player auras - only player spells can trigger player proc auras
     // Xinef: skip victim auras
+    // Excluded player shoot spells
     if (!isVictim && GetTypeId() == TYPEID_PLAYER) //spellProto->SpellFamilyName != SPELLFAMILY_GENERIC)
-        if (!(EventProcFlag & (PROC_FLAG_KILL | PROC_FLAG_DEATH)) && procSpell && procSpell->SpellFamilyName == SPELLFAMILY_GENERIC && (!eventInfo.GetTriggerAuraSpell() || eventInfo.GetTriggerAuraSpell()->SpellFamilyName == SPELLFAMILY_GENERIC))
+        if (!(EventProcFlag & (PROC_FLAG_KILL | PROC_FLAG_DEATH)) && procSpell && procSpell->SpellFamilyName == SPELLFAMILY_GENERIC && procSpell->GetCategory() != 76 &&
```
Comment (Helias) [inline src/server/game/Entities/Unit/Unit.cpp:16678]: what is `76`? We could convert it into a constant or add a comment

**Pull request merged**